src/network/networkd-route.c

```diff
@@ -2355,6 +2372,75 @@ int config_parse_multipath_route(
         return 0;
 }
 
+int config_parse_route_table_names(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(route_tables_freep) RouteTable *m = NULL;
+        _cleanup_free_ char *name = NULL;
+        OrderedHashmap **s = data;
+        uint32_t table;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                *s = ordered_hashmap_free_with_destructor(*s, route_tables_free);
+                return 0;
+        }
+
+        p = rvalue;
+        r = extract_first_word(&p, &name, ":", 0);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r <= 0 || isempty(p)) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid RouteTable=, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        r = safe_atou32(p, &table);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse RouteTable=, ignoring assignment: %s", p);
+                return 0;
+        }
+
+        if (table == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid RouteTable=, ignoring assignment: %s", p);
+                return 0;
+        }
+
+        m = new(RouteTable, 1);
+        if (!m)
+                return log_oom();
+
+        *m = (RouteTable) {
+                .table = table,
+                .name = TAKE_PTR(name),
+        };
+
+        r = ordered_hashmap_ensure_put(s, &string_hash_ops, m->name, m);
```

Comment:
Why do you use `ordered_hashmap`? I think simple `hashmap` is enough.
Also, It is not necessary to introduce `struct RoutaTable`. But just store route table number with its name as a key:
```
r = hashmap_ensure_put(s, &string_hash_ops, name, UINT32_TO_PTR(table));
```